npm_and_yarn/lib/dependabot/npm_and_yarn/package_name.rb

```diff
@@ -55,6 +73,17 @@ def scoped?
         !@scope.nil?
       end
 
+      def scoped_types_package?
+        SCOPED_TYPES_PACKAGE_REGEX.match?(to_s)
+      end
+
+      def unscoped_types_package_name
+        match = SCOPED_TYPES_PACKAGE_REGEX.match(to_s)
+        raise InvalidPackageName unless match
+
+        "@#{match[:scope]}/#{match[:name]}"
+      end
+
```

Comment:
^ From previous suggestion.

```suggestion
```

Reply:
👍 All comments addressed together in [14bb063](https://github.com/dependabot/dependabot-core/pull/5025/commits/14bb0637d9a455e903ec498a4291c88c54011571)